scripts/metricsgen/metricsdiff/metricsdiff.go

```diff
@@ -0,0 +1,156 @@
+// metricsdiff is a tool for generating a diff between two different files containing
+// prometheus metrics. metricsdiff outputs which metrics have been added, removed,
+// or have different sets of labels between the two files.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s <path> <path>
+
+Generate the diff between the sets of prometheus metrics peresent in the two files
+in the <path> files. The tool outputs which metrics have been added, removed, or
+use different label sets between the two files.
```

Comment:
A typo fix and some optional wording suggestions I thought might help.

```suggestion
Generate the diff between the two files of Prometheus metrics.
The input should have the format output by a Prometheus HTTP endpoint.
The tool indicates which metrics have been added, removed, or use different
label sets from path1 to path2.
```